bson/struct_codec.go

```diff
@@ -49,86 +47,53 @@ func (de *DecodeError) Keys() []string {
 	return reversedKeys
 }
 
-// StructCodec is the Codec used for struct values.
-//
-// Deprecated: Use [go.mongodb.org/mongo-driver/bson.NewRegistry] to get a registry with the
-// StructCodec registered.
-type StructCodec struct {
-	cache  sync.Map // map[reflect.Type]*structDescription
-	parser StructTagParser
+// mapElementsEncoder handles encoding of the values of an inline  map.
+type mapElementsEncoder interface {
+	encodeMapElements(EncodeContext, DocumentWriter, reflect.Value, func(string) bool) error
+}
+
+// structCodec is the Codec used for struct values.
+type structCodec struct {
+	cache       sync.Map // map[reflect.Type]*structDescription
+	elemEncoder mapElementsEncoder
```

Comment:
Optional: Consider using a field name that better describes the use case. E.g. `inlineMapEncoder`.